File: Assignment2.R
#note that ggplot2 had been previously installed, so the install is not present in this code
library(ggplot2)

#sets the csv to a new variable d, gets the csv from the github file
d <- read.csv("https://raw.githubusercontent.com/scvoltmer/02-DataVis-5Ways/main/cars-sample.csv")
#color maping and graph creation
sp <- ggplot(d,aes(y=MPG,x=Weight,size=Weight,color=Manufacturer)) + geom_point(alpha = 0.5) + scale_color_manual(values = c("#e1cdeb","#cacc93","#96d3bb","#90d2ef","#eabaef"))
#outputs the graph
print(sp)

File: Assignment2.py
from matplotlib import pyplot as plt
import pandas as pd


# getting the data from the forked github content
cars = pd.read_csv('https://raw.githubusercontent.com/scvoltmer/02-DataVis-5Ways/main/cars-sample.csv')

# assigning  colors to the different car companies
colors = {'ford': 'red', 'bmw': 'green', 'honda': 'blue', 'mercedes': 'yellow', 'toyota': 'purple'}

#assigning variables
x = cars['Weight']
y = cars['MPG']

# creating scatterplot
# color mapping is done using viridis, which is the built in color map for the new-ish matplotlib
plt.scatter(x, y, s=(x/25), alpha=0.3, c=cars['Manufacturer'].map(colors), cmap='viridis')
plt.xlabel("Weight")
plt.ylabel("MPG")
plt.title("Car Weight vs. Car MPG")
# showing the scatterplot
plt.show()
